numpy/core/src/multiarray/ctors.c

```diff
@@ -1390,10 +1389,12 @@ _array_from_buffer_3118(PyObject *memoryview)
 
         if (!is_ctypes) {
             /* This object has no excuse for a broken PEP3118 buffer */
-            PyErr_SetString(
+            PyErr_Format(
                     PyExc_RuntimeError,
-                    "Item size computed from the PEP 3118 buffer format "
-                    "string does not match the actual item size.");
+                    "Item size %d for PEP 3118 buffer format "
+                    "string %s does not match the dtype %c item size %d.",
```

Comment:
Better to use the `INTP` format macro here (or perhaps `%z` for itemsize, rather than casting.